Mathlib/Order/SigmaCompleteLattice.lean

```diff
@@ -0,0 +1,393 @@
+theorem σsInf_le_σsSup (hs : s.Countable) (ne : s.Nonempty) : sInf s ≤ sSup s :=
+  isGLB_le_isLUB (isGLB_σsInf hs) (isLUB_σsSup hs) ne
+
+theorem σiInf_le_σiSup [Nonempty ι] : ⨅ i, f i ≤ ⨆ i, f i :=
+  (σiInf_le (Classical.arbitrary _)).trans <| le_σiSup (Classical.arbitrary _)
+
+/-- The `sSup` of a union of two sets is the max of the suprema of each subset, under the
+assumptions that all sets are countable. -/
+theorem σsSup_union (hs : s.Countable) (ht : t.Countable) :
+    sSup (s ∪ t) = sSup s ⊔ sSup t :=
+  ((isLUB_σsSup hs).union (isLUB_σsSup ht)).σsSup_eq (hs.union ht)
+
+/-- The `sInf` of a union of two sets is the min of the infima of each subset, under the
+assumptions that all sets are countable. -/
+theorem σsInf_union (hs : s.Countable) (ht : t.Countable) :
+    sInf (s ∪ t) = sInf s ⊓ sInf t :=
+  σsSup_union (α := αᵒᵈ) hs ht
+
+/-- The supremum of an intersection of two sets is bounded above by the minimum of the suprema of
+each set, under the assumptions that all sets are countable. -/
+theorem σsSup_inter_le (hs : s.Countable) (ht : t.Countable) : sSup (s ∩ t) ≤ sSup s ⊓ sSup t :=
+  σsSup_le (hs.mono Set.inter_subset_left) fun _ hx => le_inf (le_σsSup hs hx.1) (le_σsSup ht hx.2)
+
+/-- The infimum of an intersection of two sets is bounded below by the maximum of the infima of
+each set, under the assumptions that all sets are countable. -/
+theorem le_σsInf_inter (hs : s.Countable) (ht : t.Countable) : sInf s ⊔ sInf t ≤ sInf (s ∩ t) :=
+  σsSup_inter_le (α := αᵒᵈ) hs ht
+
+/-- The supremum of `insert a s` is the maximum of `a` and the supremum of `s`, if `s` is
+nonempty and bounded above. -/
+@[simp]
+theorem σsSup_insert (hs : s.Countable) : sSup (insert a s) = a ⊔ sSup s :=
+  ((isLUB_σsSup hs).insert a).σsSup_eq <| hs.insert a
+
+/-- The infimum of `insert a s` is the minimum of `a` and the infimum of `s`, if `s` is
+nonempty and bounded below. -/
+@[simp]
+theorem σsInf_insert (hs : s.Countable) : sInf (insert a s) = a ⊓ sInf s :=
+  σsSup_insert (α := αᵒᵈ) hs
+
+theorem σsSup_le_σsSup_of_isCofinalFor (hs : s.Countable) (ht : t.Countable)
+    (hst : IsCofinalFor s t) : sSup s ≤ sSup t :=
+  IsLeast.mono (isLUB_σsSup ht) (isLUB_σsSup hs) <| upperBounds_mono_of_isCofinalFor hst
+
+theorem σsInf_le_σsInf_of_isCoinitialFor (hs : s.Countable) (ht : t.Countable)
+    (hst : IsCoinitialFor s t) : sInf t ≤ sInf s :=
+  σsSup_le_σsSup_of_isCofinalFor (α := αᵒᵈ) hs ht hst
+
+@[simp]
+theorem σsSup_empty : sSup ∅ = (⊥ : α) :=
+  (@isLUB_empty α _ _).σsSup_eq Set.countable_empty
+
+@[simp]
+theorem σsInf_empty : sInf ∅ = (⊤ : α) :=
+  σsSup_empty (α := αᵒᵈ)
+
+omit [Countable ι] in
+theorem σiSup_empty [IsEmpty ι] : ⨆ i, f i = ⊥ := by
+  simp [iSup, Set.range_eq_empty]
+
+omit [Countable ι] in
+theorem σiInf_empty [IsEmpty ι] : ⨅ i, f i = ⊤ :=
+  σiSup_empty (α := αᵒᵈ)
+
+omit [Countable ι] in
+theorem σiSup_const_mem : ⨆ (_ : ι), a ∈ ({⊥, a} : Set α) := by
```

Comment:
```suggestion
theorem σiSup_const_eq_bot_or_eq : ⨆ (_ : ι), a = ⊥ ∨ ⨆ (_ : ι), a = a := by
```
is the simp normal form and also what I'd expect to see, maybe alongside the conditions on `ι`